docs/source/deployment/databricks/databricks_notebooks_development_workflow.md

```diff
@@ -0,0 +1,293 @@
+
+Now that your project is available in a Databricks Repo, you can run it on a cluster using a notebook.
+
+To run the Python code from your Databricks repo, [create a new Python notebook](https://docs.databricks.com/notebooks/notebooks-manage.html#create-a-notebook) in your workspace. Name it `iris-databricks` for traceability and attach it to your cluster:
+
+![Create a new notebook on Databricks](../../meta/images/databricks_notebook_creation.png)
+
+### Copy project data to DBFS using dbutils
+
+On Databricks, Kedro cannot access data stored directly in your project's directory. As a result, you'll need to move your project's data to a location accessible by Databricks. You can store your project's data in the Databricks File System (DBFS), where it is accessible.
+
+A number of methods exist for moving data to DBFS. However, in this guide, you will use your new notebook and `dbutils`.
+
+To move your locally stored data to DBFS, open your `iris-databricks` notebook and in the first cell enter the following python code:
+
+```python
+dbutils.fs.cp(
+    "file:///Workspace/Repos/<databricks_username>/iris-databricks/data/",
+    "dbfs:/FileStore/iris-databricks/data",
+    recurse=True,
+)
+```
+
+Run this cell to copy the complete directory and its contents from your Repo to DBFS.
+
+To ensure that your data was copied correctly, you can list the contents of the destination directory in DBFS. Create a new cell underneath the first cell and enter the following code:
+
+```python
+dbutils.fs.ls("dbfs:/FileStore/iris-databricks/data")
+```
+
+Run this command to displays the contents of your project's `data/` directory. You can expect to see the following structure:
+
+```bash
+[FileInfo(path='dbfs:/FileStore/iris-databricks/data/01_raw', name='01_raw', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/02_intermediate', name='02_intermediate', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/03_primary', name='03_primary', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/04_feature', name='04_feature', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/05_model_input', name='05_model_input', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/06_models', name='06_models', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/07_model_output', name='07_model_output', size=...),
+ FileInfo(path='dbfs:/FileStore/iris-databricks/data/08_reporting', name='08_reporting', size=...)]
+```
+
+After these cells have successfully run, you should comment the code inside them so their operations are not unnecessarily performed during notebook runs. The cells should appear as below:
+
+**Cell 1:**
+
+```ipython
+#dbutils.fs.cp(
+#    "file:///Workspace/Repos/<databricks_username>/iris-databricks/data",
+#    "dbfs:/FileStore/iris-databricks/data",
+#    recurse=True,
+#)
+```
+
+**Cell 2:**
+
+```ipython
+#dbutils.fs.ls("dbfs:/FileStore/iris-databricks/data")
+```
```

Comment:
This is fine, so just a nitpick: is there a way to write the code in such a way that it's idempotent? Something like:

```python
if len(dbutils.fs.ls("dbfs:/FileStore/iris-databricks/data")) < 8:
    dbutils.fs.cp(...)
```

(no need to do a super exhaustive check, and it doesn't have to be this silly check of the number of files, just an idea)
```

Reply:
It's a good idea, but I wouldn't trust any simple, legible way of making the check idempotent. It would seriously confuse the user if the check returns a false positive. I would rather leave the code readable and have the user take more time to read.